connexion/exceptions.py

```diff
@@ -44,3 +44,11 @@ def __str__(self):  # pragma: no cover
 
     def __repr__(self):  # pragma: no cover
         return '<NonConformingResponse: {}>'.format(self.reason)
+
+
+class NonConformingResponseBody(NonConformingResponse):
+    pass
```

Comment:
Here to handle better all scenarios and be clear on what I am suggesting we could add:
```python
def __init__(self, reason):
   super(NonConformingResponseBody, self).__init__(
         'Response body does not conform to specification')
   self.reason = reason
```

And similar code just changing the message to `NonConformingResponseHeaders` exception.